### PR TITLE
fix(grouping): Treat in_app=None as in_app=False [ISSUE-1601]

### PR DESCRIPTION
**Pull request by jjbayer**
A customer noticed that the stack trace rule matcher `app:no` was not picked up for frames that did not have `in_app` set. This PR fixes the behavior by falling back to `in_app = False`.

### CODE REVIEW

**Review by olksdr — APPROVED** [review]
I think it makes sense

**Comment by mitsuhiko** [comment]
I'm a bit confused how this can happen. in-app not set could mean either `True` or `False` depending on the rest of the frames. I thought we run that normalization step before we apply the matchers so `in_app` should never be unset at that point.

**Comment by jjbayer** [comment]
> I'm a bit confused how this can happen. in-app not set could mean either `True` or `False` depending on the rest of the frames. I thought we run that normalization step before we apply the matchers so `in_app` should never be unset at that point.

@mitsuhiko AFAIK all the frame-modifying rules run in a single pass (there is no extra pass for in-app), which makes the outcome order-dependent. For example,

```
app:yes category=my_app
function:foo +app
```

does not set the category because the in-app flag is set only in the 2nd rule.

In this particular case, the rule is `app:no +app`. Maybe we should just disallow this type of self-referencing rules.

**Review by untitaker — APPROVED** [review]

File: src/sentry/grouping/enhancer/matchers.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ def create_match_frame(frame_data: dict, platform: Optional[str]) -> dict:
         category=get_path(frame_data, "data", "category"),
         family=get_behavior_family_for_platform(frame_data.get("platform") or platform),
         function=_get_function_name(frame_data, platform),
-        in_app=frame_data.get("in_app"),
+        in_app=frame_data.get("in_app") or False,
```
Comment (jjbayer) [inline src/sentry/grouping/enhancer/matchers.py:67]: `in_app` is explicitly None if grouping runs a second time:
https://github.com/getsentry/sentry/blob/fcb3d2ba785f55ddfb62864b564f5a379f5a522a/src/sentry/stacktraces/processing.py#L267

**Pull request merged**